hooks/useFetchStakingPoolsData.js

```diff
@@ -0,0 +1,239 @@
+  const accountBalanceOfABI = find(MultiRewards, { name: "balanceOf" });
+  const earnedABI = find(MultiRewards, { name: "earned" });
+
+  let stakingPoolsFunctions = [];
+  STAKING_POOLS.map(async (stakingPool) => {
+    stakingPoolsFunctions[stakingPool.id] = {
+      // Pool size
+      getBalanceOf: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(totalSupplyABI),
+
+      // Pool Reward For Duration
+      getRewardForDuration: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(getRewardForDurationABI),
+
+      // Last Time Reward Applicable
+      getLastTimeRewardApplicable: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(lastTimeRewardApplicableABI),
+
+      // Period Finish
+      getPeriodFinish: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(periodFinishABI),
+
+      //  Current stake
+      getAccountBalanceOf: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(accountBalanceOfABI),
+
+      // Unstaked staking token balance
+      getUnstakedBalanceOf: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(balanceOfABI),
+
+      // Unstaked staking token approval amount
+      getUnstakedAllowanceAmount: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(allowanceABI),
+
+      // Claimable token
+      getEarned: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(earnedABI),
+    };
+  });
+
+  const getPoolData = async () => {
+    return await Promise.all(
+      STAKING_POOLS.map(async (stakingPool) => {
+        // Pool size
+        const {
+          decoded: { 0: poolSize },
+        } = await stakingPoolsFunctions[stakingPool.id].getBalanceOf.call();
+
+        const rewardDataABI = find(MultiRewards, { name: "rewardData" });
+        let method =
+          connexStakingPools[stakingPool.id].rewardsContract.method(
+            rewardDataABI
+          );
+        let res = await method.call(stakingPool.rewardTokens[0].address[VECHAIN_NODE]);
+        const rewardRate = ethers.BigNumber.from(res.decoded.rewardRate);
+
+        const totalSupplyABI = find(IERC20, { name: "totalSupply" });
+        method = connex.thor
+          .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+          .method(totalSupplyABI);
+        res = await method.call();
+
+        const totalLPTokenSupply = BigNumber.from(res.decoded[0]);
+        const numberOfLPTokensStaked = poolSize;
+        // This is calculated in the amount of vex tokens instead of usd
+        const tokenTvl =
+          poolInfo[stakingPool.id].pair.tokenAmounts[0].toFixed(18);
+        // Multiply by two to get the pool TVL as this is a 50-50 pool
+        const tvlInToken = parseUnits(tokenTvl, 18).mul(2);
```

Comment:
this variable names are bit loose. `tvlInToken = tokenTvl, subtly name difference means very different thing. No need to change now but I think we need to use typescript & proxima eslint for future repos to try and make it easier for new people to come in and understand the code quickly